openhands/core/main.py

```diff
@@ -191,6 +224,27 @@ async def on_event(event: Event):
     ]:
         await asyncio.sleep(1)  # Give back control for a tick, so the agent can run
 
+    # List files in the sandbox workspace
+    # TODO(pawalt): make this recursive
+    files = runtime.list_files(config.workspace_mount_path_in_sandbox)
+
+    # Read contents of each file and write to workspace_base
+    for file in files:
+        sandbox_file_path = os.path.join(config.workspace_mount_path_in_sandbox, file)
+        read_action = FileReadAction(path=sandbox_file_path)
+        observation = runtime.read(read_action)
+
+        if isinstance(observation, ErrorObservation):
+            logger.error(f'Failed to read file {file}: {observation.message}')
+        else:
+            logger.info(f'Contents of {file}:')
+            logger.info(observation.content)
+
+            # Write the content to the file in workspace_base
+            workspace_file_path = os.path.join(config.workspace_base, file)
+            with open(workspace_file_path, 'w') as f:
+                f.write(observation.content)
```

Comment:
Ditto here, let's remove this, and make the user responsible for doing a `git pull` and `git push`, or downloading their work.

Forthcoming UX improvements will make this a much better experience

Reply:
Sounds good so long as we're good re: integration tests not working